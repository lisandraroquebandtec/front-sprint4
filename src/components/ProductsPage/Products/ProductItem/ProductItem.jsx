import {
    ProductsCard,
    Card,
    CardImg,
    CardDescription,
    CardPrice,
  } from "./StyleProductItem";

function ProductItem({image, name, price}){
    return(
        <ProductsCard>
              <Card>
                <CardImg src={image} />
                <CardDescription>{name}</CardDescription>
                <CardPrice>{price}</CardPrice>
              </Card>
        </ProductsCard>
    );
}

export default ProductItem;
