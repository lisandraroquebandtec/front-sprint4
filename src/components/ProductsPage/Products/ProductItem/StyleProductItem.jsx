import styled from "styled-components";

export const ProductsCard = styled.li`
  display: flex;
  justify-content: space-between;
`;

export const Card = styled.div`
  cursor: pointer;
  border: 1px solid white;
  padding: 8px;
  max-width: 250px;
`;

export const CardImg = styled.img`
  max-width: 100%;
  max-height: 100%;
`;

export const CardDescription = styled.p`
  font-size: 0.6875rem;
  font-weight: 400;
  color: rgba(0, 0, 0, 0.87);
  text-transform: uppercase;
  text-align: center;
  margin-top: 8px;
  margin-bottom: 10px;
`;

export const CardPrice = styled.p`
  font-size: 0.8125rem;
  font-weight: 600;
  text-transform: uppercase;
  text-align: center;
`;
