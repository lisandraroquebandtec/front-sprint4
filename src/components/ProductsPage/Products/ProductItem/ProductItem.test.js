import React from 'react';
import { render} from '@testing-library/react';
import ProductItem from './ProductItem';

describe('O componente de item do produto,', () =>{
  it('o snapshot do componente deve permanecer sempre o mesmo', ()=>{
    const {container} = render(
    <ProductItem
    image='/assets/images/c1.webp'
    name='Jaqueta Puffer Juvenil com Capuz Super Mario Branco Tam 10 a 18'
    price='199,90'
    />)
    expect(container.firstChild).toMatchSnapshot();
  })
})