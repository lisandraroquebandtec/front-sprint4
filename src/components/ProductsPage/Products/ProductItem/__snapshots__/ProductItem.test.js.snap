// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`O componente de item do produto, o snapshot do componente deve permanecer sempre o mesmo 1`] = `
<li
  class="sc-bdnxRM ctBBVd"
>
  <div
    class="sc-gtsrHT EKXoD"
  >
    <img
      class="sc-dlnjwi fYWFt"
      src="/assets/images/c1.webp"
    />
    <p
      class="sc-hKFxyN hdqroc"
    >
      Jaqueta Puffer Juvenil com Capuz Super Mario Branco Tam 10 a 18
    </p>
    <p
      class="sc-eCApnc rvNhV"
    >
      R$ 
      199,90
    </p>
  </div>
</li>
`;
